azure_pipeline.yaml

```diff
@@ -13,7 +13,58 @@ variables:
   action:
 
 stages:
+  - stage: Precheck
+    jobs:
+      - job: Validate
+        pool:
+          vmImage: $(agentPool)
+        steps:
+        - task: AzureKeyVault@1
+          inputs:
+            ConnectedServiceName: azurerm-sandbox
+            keyVaultName: infra-vault-nonprod
+            secretsFilter: 'azure-devops-token'
+            runAsPreJob: false
+        - task: Bash@3
+          env:
+            thisbuild: $(Build.BuildId)
+          inputs:
+            targetType: 'inline'
+            script: | 
+              #!/bin/bash
+              set -e
+
+              organization="hmcts"
+              project="CNP"
+
+              echo "This build number is: $thisbuild"
+
+              while [[ $thisbuild -ne $buildrunning ]]
+              do
+              JSON_DATA=$(curl -u :$(azure-devops-token) --request GET "https://dev.azure.com/$organization/$project/_apis/build/builds?api-version=5.1&definitions=235" -H "Content-Type: application/json" | jq  '.value[] | .status + (.id|tostring)' | sort -u)
```

Comment:
extract pipeline definition to a variable as well

Reply:
Cheers - now updated